protocol/signalfx/signalfx.go

```diff
@@ -29,6 +29,9 @@ func NewDatumValue(val sfxmodel.Datum) datapoint.Value {
 
 // ValueToValue converts the v2 JSON value to a core api Value
 func ValueToValue(v ValueToSend) (datapoint.Value, error) {
+	if v == nil {
```

Comment:
can we instead do if v != nil { } as it return error at the end anyways?

Reply:
Sure. Old habits of kernel development with early returns on bad inputs. 😄